torch/_inductor/fx_passes/freezing_patterns.py

```diff
@@ -102,6 +102,8 @@ def lazy_init():
 
 
 def register_freezing_graph_pattern(pattern, extra_check=_return_true, pass_number=0):
+    while pass_number > len(pass_patterns) - 1:
+        pass_patterns.append(PatternMatcherPass())
```

Comment:
can we simplify this with list extend?

Reply:
Do you mean extending list length explicitly?
From
```
pass_patterns = [
    PatternMatcherPass(),
    PatternMatcherPass(),
    PatternMatcherPass(),
]
```
to
```
pass_patterns = [
    PatternMatcherPass(),
    PatternMatcherPass(),
    PatternMatcherPass(),
    PatternMatcherPass(),
    PatternMatcherPass(),
    PatternMatcherPass(),
]
```

Reply:
I want to have a lazy init so I extend the list here. I thought it might be better.

Reply:
I meant invoking `pass_patterns.extend` here.

Reply:
I see. However, with `extend` the code would look like below, which looks less simplified...
```python
    if pass_number > len(pass_patterns) - 1:
        pass_patterns.extend([PatternMatcherPass()] * (pass_number - len(pass_patterns) + 1))
```